src/components/SearchBar/SearchBar.tsx

```diff
@@ -1,10 +1,42 @@
 import React from 'react';
 import styles from './searchbar.module.scss';
+import { searchMovies } from '../../services/movie';
+import { TMovieCard } from '../../types/TypeMovieDB';
 
-const SearchBar = () => {
+type TProps = {
+  setMovies: React.Dispatch<React.SetStateAction<TMovieCard[]>>;
+  query: string;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+};
+
+const SearchBar: React.FC<TProps> = (props: TProps) => {
+  const searchMovie = async (e: { preventDefault: () => void }) => {
```

Comment:
burada e'nin tipini `HTMLInputElement` olarak verirsen daha doğru olur. sonrasında `e.target.value` vs gibi değerlere de erişebilirsin

Reply:
Burası onChange olduğu için HTMLInputElement kabul etmiyor veya nasıl yapmalıyım. 
Burada düzelttim
 const changeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
    props.setQuery(e.target.value);
  };